### PR TITLE
Remove unused cimports, fix #22067

### LINKED ISSUE #22067: TypeError in pandas._libs.tslibs.period._ordinal_from_fields
Using pandas `0.23.3` I receive a `TypeErorr` when running the following code:

```python
import pandas
ts = pandas.date_range('2017-01-01', '2018-01-01', freq='B')
months = ts.map(lambda x: pandas.Period(year=x.year, month=x.month, freq='M'))
# Screen output even though operation succeeds.
# TypeError                                 Traceback (most recent call last)
# TypeError: an integer is required
# Exception TypeError: 'an integer is required' in 'pandas._libs.tslibs.period._ordinal_from_fields' ignored
```
Oddly, enough the operation succeeds, months has the correct data.
#### Expected Output
I would expect no output to stderr or stdout, just the correct assignment of the map operation. This is more of an annoyance than anything else.

#### Output of ``pd.show_versions()``
```
pandas.show_versions()

INSTALLED VERSIONS
------------------
commit: None
python: 2.7.15.final.0
python-bits: 64
OS: Darwin
OS-release: 17.7.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: None.None

pandas: 0.23.3
pytest: None
pip: 18.0
setuptools: 40.0.0
Cython: None
numpy: 1.13.0
scipy: 1.1.0
pyarrow: None
xarray: None
IPython: 5.7.0
sphinx: None
patsy: None
dateutil: 2.7.3
pytz: 2018.5
blosc: None
bottleneck: None
tables: 3.4.4
numexpr: 2.6.6
feather: None
matplotlib: 2.2.2
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: 1.0.1
sqlalchemy: 1.2.10
pymysql: 0.9.2
psycopg2: None
jinja2: 2.10
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None
```
<details>

[paste the output of ``pd.show_versions()`` here below this line]

</details>

### PR DESCRIPTION
**Pull request by jbrockmendel**
Fixes #22067, but no clear way to test it.

- [x] closes #22067

### CODE REVIEW

File: pandas/_libs/tslibs/period.pyx [hunk 1]
```diff
@@ -1920,8 +1920,8 @@ class Period(_Period):
         return cls._from_ordinal(ordinal, freq)
 
 
-cdef int64_t _ordinal_from_fields(year, month, quarter, day,
-                                  hour, minute, second, freq):
+cdef int64_t _ordinal_from_fields(int year, int month, quarter, int day,
+                                  int hour, int minute, int second, freq):
```
Comment (jbrockmendel) [inline pandas/_libs/tslibs/period.pyx]: This is what fixed 22067.  year and month were getting an Int64Index.

File: pandas/_libs/tslibs/period.pyx [hunk 2]
```diff
@@ -1920,8 +1920,8 @@ class Period(_Period):
         return cls._from_ordinal(ordinal, freq)
 
 
-cdef int64_t _ordinal_from_fields(year, month, quarter, day,
-                                  hour, minute, second, freq):
+cdef int64_t _ordinal_from_fields(int year, int month, quarter, int day,
```
Comment (jreback) [inline pandas/_libs/tslibs/period.pyx]: does quarter not need a type?

**Review by jreback — CHANGES_REQUESTED** [review]
can you add a test for #22067 ?

**Comment by jreback** [comment]
minor comments and pls rebase

File: pandas/tests/indexes/datetimes/test_datetime.py [hunk 3]
```diff
@@ -126,6 +127,22 @@ def test_map(self):
         exp = Index([f(x) for x in rng], dtype='<U8')
         tm.assert_index_equal(result, exp)
 
+    def test_map_fallthrough(self):
+        # GH#22067, check we don't get warnings about silently ignored errors
+        dti = date_range('2017-01-01', '2018-01-01', freq='B')
+
+        c = StringIO()
```
Comment (jreback) [inline pandas/tests/indexes/datetimes/test_datetime.py]: use ``tm.capture_stderr``

**Review by jreback — CHANGES_REQUESTED** [review]
rebase & comments, lgtm. ping on green.

**Comment by jbrockmendel** [comment]
Ping

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks!

**Pull request merged**